app/src/main/java/com/duckduckgo/app/browser/BrowserTrackersAnimatorHelper.kt

```diff
@@ -74,33 +79,92 @@ class BrowserTrackersAnimatorHelper {
         return createLogosViewList(activity, container, logos)
     }
 
-    private fun createLogosViewList(activity: Activity, container: ConstraintLayout, resourcesId: List<Int>?): List<View> {
+    private fun createLogosViewList(
+        activity: Activity,
+        container: ConstraintLayout,
+        resourcesId: List<TrackerLogo>?
+    ): List<View> {
         return resourcesId?.map {
-            val imageView = ImageView(activity)
-            imageView.scaleType = ImageView.ScaleType.CENTER_CROP
-            imageView.setImageResource(R.drawable.network_cross_anim)
-            imageView.setBackgroundResource(it)
-            imageView.id = View.generateViewId()
-            container.addView(imageView)
-            return@map imageView
+            return@map if (it.resId == R.drawable.other_tracker_bg) {
+                val frameLayout = createTrackerTextLogo(activity, it)
+                container.addView(frameLayout)
+                frameLayout
+            } else {
+                val imageView = createTrackerImageLogo(activity, it)
+                container.addView(imageView)
+                imageView
+            }
         }.orEmpty()
     }
 
-    private fun createResourcesIdList(activity: Activity, events: List<TrackingEvent>): List<Int>? {
+    private fun createTrackerTextLogo(activity: Activity, trackerLogo: TrackerLogo): FrameLayout {
+        val params = FrameLayout.LayoutParams(FrameLayout.LayoutParams.WRAP_CONTENT, FrameLayout.LayoutParams.WRAP_CONTENT)
+        val frameLayout = FrameLayout(activity)
+        frameLayout.id = View.generateViewId()
+
+        val animationView = ImageView(activity)
+        animationView.setImageResource(R.drawable.network_cross_anim)
+        animationView.layoutParams = params
+
+        val textView = AppCompatTextView(activity)
+        textView.gravity = Gravity.CENTER
+        textView.setBackgroundResource(trackerLogo.resId)
+        TextViewCompat.setTextAppearance(textView, R.style.UnknownTrackerText)
+        textView.text = trackerLogo.trackerLetter
+        textView.layoutParams = params
+
+        frameLayout.addView(textView)
+        frameLayout.addView(animationView)
+
+        return frameLayout
+    }
+
+    private fun createTrackerImageLogo(activity: Activity, trackerLogo: TrackerLogo): ImageView {
+        val imageView = ImageView(activity)
+        imageView.scaleType = ImageView.ScaleType.CENTER_CROP
+        imageView.setImageResource(R.drawable.network_cross_anim)
+        imageView.setBackgroundResource(trackerLogo.resId)
+        imageView.id = View.generateViewId()
+        return imageView
+    }
+
+    private fun createResourcesIdList(activity: Activity, events: List<TrackingEvent>): List<TrackerLogo>? {
         if (activity.packageName == null) return emptyList()
+        val resourcesList = events.asSequence()
+            .mapNotNull { it.entity }
+            .sortedByDescending { it.prevalence }
+            .map {
```

Comment:
NIT: knowing that some sites have more than 30 trackers and we will show just 3 (but we need to know if there are more). What about mapping only 4 events?

Reply:
Done!